### PR TITLE
XOR/standard nonlocal game converter

### PR DESCRIPTION
**Pull request by juliusw352**
## Description
Created a brief function to convert the 2-dimensional predicate matrix of an XOR game into a 4-dimensional one compatible with the `NonlocalGame` object.

## Questions
-  [x] I am not entirely sure how to test this function properly. For now, I'm just creating a new `NonlocalGame` and check if the classical values are the same. Is there a better, more reliable testing method?

## Other
This is a preliminary step to solve issue #15, the approach being to take advantage of already existing features in the `NonlocalGame` class by enabling access to them from the `XORGame` class.

## Status
-  [x] Ready to go

### CODE REVIEW

File: tests/test_nonlocal_games/test_xor_game.py [hunk 1]
```diff
@@ -168,6 +169,18 @@ def test_zero_prob_mat(self):
             game = XORGame(prob_mat, pred_mat)
             game.quantum_value()
 
+    def test_to_nonlocal_game(self):
+        prob = np.array([[0.25,0.25],[0.25,0.25]])
```
Comment (vprusso) [inline tests/test_nonlocal_games/test_xor_game.py:173]: We could simplify this slightly by:

```python
prob = 1/4 * np.ones((2, 2))
```

File: tests/test_nonlocal_games/test_xor_game.py [hunk 2]
```diff
@@ -168,6 +169,18 @@ def test_zero_prob_mat(self):
...
+        pred = np.array([[0,0],[0,1]])
+
+        specific_game = XORGame(pred, prob, 1)
```
Comment (vprusso) [inline tests/test_nonlocal_games/test_xor_game.py:176]: Perhaps, to make the names a bit more explicit, we could alter these lines to:

```python
xor_chsh = XORGame(prob, pred, reps=1)
nlg_pred = xor_chsh.to_nonlocal_game()

nlg_chsh = NonlocalGame(prob, nlg_pred, reps=1)

np.testing.assert_equal(nlg_chsh.classical_value(), xor_chsh.classical_value())
```

File: tests/test_nonlocal_games/test_xor_game.py [hunk 3]
```diff
@@ -168,6 +169,18 @@ def test_zero_prob_mat(self):
...
+
+        generic_pred = specific_game.to_nonlocal_game()
+        generic_game = NonlocalGame(prob, generic_pred, 1)
+
+
```
Comment (vprusso) [inline tests/test_nonlocal_games/test_xor_game.py:181]: Extra space here.

File: toqito/nonlocal_games/xor_game.py [hunk 4]
```diff
@@ -290,3 +290,20 @@ def classical_value(self) -> float:
             "multiple repetitions for the classical value of "
             "a nonlocal game."
         )
+    def to_nonlocal_game(self) -> np.ndarray:
```
Comment (vprusso) [inline toqito/nonlocal_games/xor_game.py:293]: I wonder if we want to refactor this to return a `NonlocalGame` object. 

So:

```python
        q_0, q_1 = self.prob_mat.shape
        pred_mat = self.pred_mat
        xor_pred_mat = np.ndarray((2, 2, q_0, q_1))

        for a in range(2):
            for b in range(2):
                for x in range(q_0):
                    for y in range(q_1):
                        xor_pred_mat[a, b, x, y] = pred_mat[x, y] == a ^ b

        return NonlocalGame(self.prob_mat, xor_pred_mat, reps=self.reps)
```

*Note*: Haven't tested the above, but something like this

**Review by vprusso — COMMENTED** [review]
Great pass at this @juliusw352 !

I added some comments here, but I think they are fairly minor. Thank you yet again for your contribution to the `toqito` project!

**Comment by juliusw352** [comment]
Hi,

I have made the suggested changes, and testing still passes.
Let me know if you are happy with the changes made, and I'll mark the PR as ready to go.

Cheers,
Julius

**Review by vprusso — APPROVED** [review]

**Comment by vprusso** [comment]
Looks good to me. Merge when ready! 🚀 

Thank you again for contributing to the `toqito` project! I hope to see you around again and hope to see you for the upcoming unitaryHack project! Thank you again, @juliusw352 !

**Comment by juliusw352** [comment]
Thank you! I made one final change because I forgot to update the documentation after changing the return type to a `NonlocalGame` object.

**Pull request merged**